src/ReadoutApplication.cpp

```diff
@@ -205,65 +211,97 @@ ReadoutApplication::generate_modules(oksdbinterfaces::Configuration* confdb,
       TLOG_DEBUG(7) << "Ignoring disabled ReadoutGroup " << roGroup->UID();
       continue;
     }
+    // get the readout groups and the interfaces and streams therein; 1 reaout group corresponds to 1 data reader module
     auto rset = roGroup->cast<coredal::ReadoutGroup>();
+
     if (rset == nullptr) {
         throw (BadConf(ERS_HERE, "ReadoutApplication contains something other than ReadoutGroup"));
     }
     std::vector<const coredal::Connection*> outputQueues;
-    for (auto res : rset->get_contains()) {
-      auto stream = res->cast<coredal::DROStreamConf>();
-      if (stream == nullptr) {
-        throw (BadConf(ERS_HERE, "ReadoutGroup contains something other than DROStreamConf"));
-      }
-      if (stream->disabled(*session)) {
-        TLOG_DEBUG(7) << "Ignoring disabled DROStreamConf " << stream->UID();
+    if (rset->get_contains().empty()) {
+        throw (BadConf(ERS_HERE, "ReadoutGroup does not contain interfaces"));
+    }
+
+    std::vector<const oksdbinterfaces::ConfigObject*> ifObjs;
+    auto interfaces = rset->get_contains();
+    for (auto res_set : interfaces) {
```

Comment:
Maybe `rset` and `res_set` are not the best names. Perhaps `group_rset` and `interface_rset' would be better.